### PR TITLE
Adding query option information to get cred thumbprint info

### PR DESCRIPTION
**Pull request by madansr7**

### CODE REVIEW

**Comment by madansr7** [comment]
@FaithOmbongi can you please review.

File: api-reference/beta/api/application-get.md [hunk 1]
```diff
@@ -30,7 +30,13 @@ One of the following permissions is required to call this API. To learn more, in
 GET /applications/{id}
 ```
 ## Optional query parameters
+
 This method supports the `$select` [OData query parameter](/graph/query-parameters) to retrieve specific application properties.
+By default the api does not return the keyCredential thumbprint value when listing all applications. The thumbprint value (`key` property) for application keyCredentials object is returned only if keyCredentials are explicitly requested using a $select when listing all applications or when a single application is queried.
+For example, to get public key thumbprint information for all applications, use the following query `$select=id,appId,keyCredentials`.
+
+The use of $select to get keyCredentials when listing all applications has a throttling limit of 150 requests per minute for every tenant.
+
```
Comment (FaithOmbongi) [inline api-reference/beta/api/application-get.md:39]:
```suggestion

By default, this API doesn't return the value of the **key** thumbprint in the **keyCredentials** property unless **keyCredentials** is specified in a `$select` query. For example, `$select=id,appId,keyCredentials`.

The use of `$select` to get **keyCredentials** for applications has a throttling limit of 150 requests per minute for every tenant.

```

File: api-reference/beta/api/application-list.md [hunk 2]
```diff
@@ -38,6 +38,11 @@ GET /applications
 
 This method supports the `$count`, `$expand`, `$filter`, `$orderBy`, `$search`, `$select`, and `$top` [OData query parameters](/graph/query-parameters) to help customize the response. Some queries are supported only when you use the **ConsistencyLevel** header set to `eventual` and `$count`. For more information, see [Advanced query capabilities on Azure AD directory objects](/graph/aad-advanced-queries).
 
+By default the api does not return the keyCredential thumbprint value when listing all applications. The thumbprint value (`key` property) for application keyCredentials object is returned only if keyCredentials are explicitly requested using a $select when listing all applications or when a single application is queried.
+For example, to get public key thumbprint information for all applications, use the following query `$select=id,appId,keyCredentials`.
+
+The use of $select to get keyCredentials when listing all applications has a throttling limit of 150 requests per minute for every tenant.
+
```
Comment (FaithOmbongi) [inline api-reference/beta/api/application-list.md:45]:
```suggestion

By default, this API doesn't return the value of the **key** thumbprint in the **keyCredentials** property when listing all applications. To retrieve the **key** thumbprint, the **keyCredentials** property must be specified in a `$select` query. For example, `$select=id,appId,keyCredentials`.

The use of `$select` to get **keyCredentials** for applications has a throttling limit of 150 requests per minute for every tenant.

```

File: api-reference/beta/api/serviceprincipal-get.md [hunk 3]
```diff
@@ -34,6 +34,13 @@ GET /servicePrincipals/{id}
 ## Optional query parameters
 This method supports the [OData Query Parameters](/graph/query-parameters) to help customize the response.
 
+This method supports the `$count`, `$expand`, `$filter`, `$orderBy`, `$search`, `$select`, and `$top` [OData query parameters](/graph/query-parameters) to help customize the response. Some queries are supported only when you use the **ConsistencyLevel** header set to `eventual` and `$count`. For more information, see [Advanced query capabilities on Azure AD directory objects](/graph/aad-advanced-queries).
+
+By default the api does not return the keyCredential thumbprint value when listing all service principals. The thumbprint value (`key` property) for service principal keyCredentials object is returned only if keyCredentials are explicitly requested using a $select when listing all service principals or when a single service principal is queried.
+For example, to get public key thumbprint information for all service principals, use the following query `$select=id,appId,keyCredentials`.
+
+The use of $select to get keyCredentials when listing all service principals has a throttling limit of 150 requests per minute for every tenant.
+
```
Comment (FaithOmbongi) [inline api-reference/beta/api/serviceprincipal-get.md:43]:
```suggestion
By default, this API doesn't return the value of the **key** thumbprint in the **keyCredentials** property unless **keyCredentials** is specified in a `$select` query. For example, `$select=id,appId,keyCredentials`.

The use of `$select` to get **keyCredentials** for service principals has a throttling limit of 150 requests per minute for every tenant.

```

File: api-reference/beta/api/serviceprincipal-list.md [hunk 4]
```diff
@@ -36,6 +36,11 @@ GET /servicePrincipals
 
 This method supports the `$count`, `$expand`, `$filter`, `$orderBy`, `$search`, `$select`, and `$top` [OData query parameters](/graph/query-parameters) to help customize the response. Some queries are supported only when you use the **ConsistencyLevel** header set to `eventual` and `$count`. For more information, see [Advanced query capabilities on Azure AD directory objects](/graph/aad-advanced-queries).
 
+By default the api does not return the keyCredential thumbprint value when listing all service principals. The thumbprint value (`key` property) for service principal keyCredentials object is returned only if keyCredentials are explicitly requested using a $select when listing all service principals or when a single service principal is queried.
+For example, to get public key thumbprint information for all service principals, use the following query `$select=id,appId,keyCredentials`.
+
+The use of $select to get keyCredentials when listing all service principals has a throttling limit of 150 requests per minute for every tenant.
+
```
Comment (FaithOmbongi) [inline api-reference/beta/api/serviceprincipal-list.md:43]:
```suggestion
By default, this API doesn't return the value of the **key** thumbprint in the **keyCredentials** property when listing all applications. To retrieve the **key** thumbprint, the **keyCredentials** property must be specified in a `$select` query. For example, `$select=id,appId,keyCredentials`.

The use of `$select` to get **keyCredentials** for service principals has a throttling limit of 150 requests per minute for every tenant.

```

File: api-reference/v1.0/api/application-get.md [hunk 5]
```diff
@@ -27,21 +27,31 @@ One of the following permissions is required to call this API. To learn more, in
...
 ```
+
 ## Optional query parameters
-This method supports the `$select` [OData query parameter](/graph/query-parameters) to retrieve specific application properties. 
+
+This method supports the `$select` [OData query parameter](/graph/query-parameters) to retrieve specific application properties.
+By default the api does not return the keyCredential thumbprint value when listing all applications. The thumbprint value (`key` property) for application keyCredentials object is returned only if keyCredentials are explicitly requested using a $select when listing all applications or when a single application is queried.
+For example, to get public key thumbprint information for all applications, use the following query `$select=id,appId,keyCredentials`.
+
+The use of $select to get keyCredentials when listing all applications has a throttling limit of 150 requests per minute for every tenant.
```
Comment (FaithOmbongi) [inline api-reference/v1.0/api/application-get.md:37]:
```suggestion

By default, this API doesn't return the value of the **key** thumbprint in the **keyCredential** property unless **keyCredentials** is specified in a `$select` query. For example, `$select=id,appId,keyCredentials`.

The use of `$select` to get **keyCredentials** for applications has a throttling limit of 150 requests per minute for every tenant.
```

File: api-reference/v1.0/api/application-list.md [hunk 6]
```diff
@@ -35,6 +35,13 @@ GET /applications
...
 
+This method supports the `$count`, `$expand`, `$filter`, `$orderBy`, `$search`, `$select`, and `$top` [OData query parameters](/graph/query-parameters) to help customize the response. Some queries are supported only when you use the **ConsistencyLevel** header set to `eventual` and `$count`. For more information, see [Advanced query capabilities on Azure AD directory objects](/graph/aad-advanced-queries).
+
+By default the api does not return the keyCredential thumbprint value when listing all applications. The thumbprint value (`key` property) for application keyCredentials object is returned only if keyCredentials are explicitly requested using a $select when listing all applications or when a single application is queried.
+For example, to get public key thumbprint information for all applications, use the following query `$select=id,appId,keyCredentials`.
+
+The use of $select to get keyCredentials when listing all applications has a throttling limit of 150 requests per minute for every tenant.
+
```
Comment (FaithOmbongi) [inline api-reference/v1.0/api/application-list.md:44]:
```suggestion

By default, this API doesn't return the value of the **key** thumbprint in the **keyCredentials** property when listing all applications. To retrieve the **key** thumbprint, the **keyCredentials** property must be specified in a `$select` query. For example, `$select=id,appId,keyCredentials`.

The use of `$select` to get **keyCredentials** for applications has a throttling limit of 150 requests per minute for every tenant.

```

File: api-reference/v1.0/api/serviceprincipal-get.md [hunk 7]
```diff
@@ -31,6 +31,11 @@ GET /servicePrincipals/{id}
 ## Optional query parameters
 This method supports the [OData Query Parameters](/graph/query-parameters) to help customize the response.
 
+By default the api does not return the keyCredential thumbprint value when listing all service principals. The thumbprint value (`key` property) for service principal keyCredentials object is returned only if keyCredentials are explicitly requested using a $select when listing all service principals or when a single service principal is queried.
+For example, to get public key thumbprint information for all service principals, use the following query `$select=id,appId,keyCredentials`.
+
+The use of $select to get keyCredentials when listing all service principals has a throttling limit of 150 requests per minute for every tenant.
+
```
Comment (FaithOmbongi) [inline api-reference/v1.0/api/serviceprincipal-get.md:38]:
```suggestion
By default, this API doesn't return the value of the **key** thumbprint in the **keyCredentials** property unless **keyCredentials** is specified in a `$select` query. For example, `$select=id,appId,keyCredentials`.

The use of `$select` to get **keyCredentials** for service principals has a throttling limit of 150 requests per minute for every tenant.

```

File: api-reference/v1.0/api/serviceprincipal-list.md [hunk 8]
```diff
@@ -33,6 +33,11 @@ GET /servicePrincipals
 
 This method supports the `$count`, `$expand`, `$filter`, `$orderBy`, `$search`, `$select`, and `$top` [OData query parameters](/graph/query-parameters) to help customize the response. Some queries are supported only when you use the **ConsistencyLevel** header set to `eventual` and `$count`. For more information, see [Advanced query capabilities on Azure AD directory objects](/graph/aad-advanced-queries).
 
+By default the api does not return the keyCredential thumbprint value when listing all service principals. The thumbprint value (`key` property) for service principal keyCredentials object is returned only if keyCredentials are explicitly requested using a $select when listing all service principals or when a single service principal is queried.
+For example, to get public key thumbprint information for all service principals, use the following query `$select=id,appId,keyCredentials`.
+
+The use of $select to get keyCredentials when listing all service principals has a throttling limit of 150 requests per minute for every tenant.
+
```
Comment (FaithOmbongi) [inline api-reference/v1.0/api/serviceprincipal-list.md:40]:
```suggestion
By default, this API doesn't return the value of the **key** thumbprint in the **keyCredentials** property when listing all applications. To retrieve the **key** thumbprint, the **keyCredentials** property must be specified in a `$select` query. For example, `$select=id,appId,keyCredentials`.

The use of `$select` to get **keyCredentials** for service principals has a throttling limit of 150 requests per minute for every tenant.

```

**Review by FaithOmbongi — APPROVED** [review]
- Tweaked the sentences a bit.
- Also proposing you include a note in the [throttling guidance](https://docs.microsoft.com/en-us/graph/throttling?context=graph%2Fapi%2F1.0&view=graph-rest-1.0#pattern) about the cost of `$select` queries on application throttling limits.

Approved with minor changes requested.

**Review by FaithOmbongi — APPROVED** [review]

**Comment by madansr7** [comment]
@Lauragra @FaithOmbongi can we please get this merged.

**Comment by FaithOmbongi** [comment]
@madansr7 , please remember to add the **ready to merge** label so that Laura can see it in her merge queue.

**Pull request merged**